opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerGuru.java

```diff
@@ -261,7 +262,7 @@ public class AnalyzerGuru {
                 new IgnorantAnalyzerFactory(),
                 new BZip2AnalyzerFactory(),
                 new XMLAnalyzerFactory(),
-                YamlAnalyzerFactory.DEFAULT_INSTANCE,
+                new YamlAnalyzerFactory(),
```

Comment:
why is this needed ?

Reply:
Removed this singleton instances are harder to test, so it is recommended to avoid singleton design pattern. 
https://rules.sonarsource.com/java/RSPEC-6548/